lib/pages/news_feed/view/news_feed_page.dart

```diff
@@ -24,6 +24,23 @@ class NewsFeedPage extends StatelessWidget {
           }
 
           final List<NewsFeedItem> newsFeedItems = snapshot.data;
+          if (newsFeedItems == null) {
+            return Center(
+                child: Padding(
+                    padding: const EdgeInsets.all(8),
+                    child: Text(
+                      S.of(context).warningUnableToReachNewsFeed,
+                      textAlign: TextAlign.center,
+                      style: TextStyle(color: Theme.of(context).disabledColor),
+                    )));
+          } else if (newsFeedItems.isEmpty) {
+            return Center(

```

Comment:
If it goes to this case, it'll be a very empty and ugly page. Show an illustration and a more specific message (see, for example, what's shown on the classes page when the user hasn't added any classes).